src/irr/builtin/embedResource.py

```diff
@@ -0,0 +1,51 @@
+# takes a single argument {filename}
+# the variable name inside the generated file is always constant - resource, resource_len
+# the output file is located next to the original, its name is similiar to the input file,
+# but ends with .h
+
+import sys
+
+# for getting the file size
+# import os
+
+# for regex to truncate extensions
+import re
+
+if  len(sys.argv) < 2 :
+    print("Incorrect argument count")
+else:
+
+    #arguments
+    inputfilename = sys.argv[1]
+    outputfilename = inputfilename + '.h'
+    #outputfilename = re.sub(r'\..+', '', inputfilename)+ '.h'
+
+    #file size
+    # fsize = os.stat(inputfilename).st_size
+
+    #opening a file
+    outp = open(outputfilename,"w+")
+
+    outp.write("#include <stdlib.h>\n")
+
+    # outp.write("const uint8_t* resource[%d] = {\n" % fsize) # array size is not required
+    outp.write("const uint8_t* resource[] = {\n")
```

Comment:
you need a unique variable name instead of `resource`

I'd take the original virtual path like `irr/builtin/X/Y/Z/something.type` and I would replace the `/` or `\` and `.` with `_` and use that as a name, for now